Added loginshell config file #4620

sec65 opened the pull request:
# Description

If nushell is used as login-shell (-l) and a login.nu file is present in the standard config directory, the file is run. The file is not created (or copied from a default) if it doen't exist unlike the other config files.
Fixes #4620

# Tests

Make sure you've run and fixed any issues with these commands:

- [x] `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- [x] `cargo clippy --workspace --features=extra -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- [x] `cargo test --workspace --features=extra` to check that all the tests pass

fdncred commented:
can you please add an example login.nu script so people understand what it's for? it should probably go in the docs/sample_config folder. it doesn't have to be anything fancy.

also, I'm not clear yet on where this falls in the execution order. I didn't study the changes real close. for instance, right now, env.nu gets sourced first and those variables are available to config.nu. where does login.nu get sourced in the timeline?

sec65 commented:
Added a sample login.nu script 

At the moment the login.nu will be sourced after config/env.nu, since I assumed it should be possible to change some of the config variables set in config/env.nu in case of a loginshell.

fdncred commented:
Sorry to have more questions but I was also wondering if perhaps we should have a `$nu.loginshell-path` like we have `$nu.config-path` and `$nu.env-path`. What do you think?

sec65 commented:
Sure, I can add a `nu.loginshell-path` variable, no problem. But as far as I have seen, there is no information whether it's a loginshell or not in `eval_variable` (in eval.rs), so the variable would be set in any case. Or maybe I'm missing something?

fdncred commented:
ya, I think the $nu.loginshell-path would be set no matter if you're using it as a login shell or not.

kubouch commented:
I don't have any objections but the configuration is getting pretty complex at this point (env.nu, config.nu, login.nu). I think it's important to also add a section to the documentation (https://www.nushell.sh/book/configuration.html#nushell-configuration-with-env-nu-and-config-nu), explain what a login shell is, and list the exact loading order in which the config files are run.

And yeah, `$nu.loginshell-path` can be set permanently, even if the file doesn't exist, I think it's fine.

sec65 commented:
Thanks for the response. I added the variable `$nu.loginshell-path` and added a section in the docu (see https://github.com/nushell/nushell.github.io/pull/488).

fdncred commented:
Thanks for sticking with us. This looks good to me.
